source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -642,14 +642,14 @@ protected async Task<TResult[]> ReadResultsAsync<TResult>(PreparedCommand prepar
             if (mapping.IdColumn?.Type != typeof(TKey))
                 throw new ArgumentException($"Provided Id of type '{typeof(TKey).FullName}' does not match configured type of '{mapping.IdColumn?.Type.FullName}'.");
 
-            var columnNames = GetColumnNames(mapping.SchemaName, mapping.TableName);
+            var columnNames = GetColumnNames(configuration.GetSchemaNameOrDefault(mapping), mapping.TableName);
             var param = new CommandParameterValues();
             param.AddTable("criteriaTable", idList.ToList(), configuration);
             var statement = $"SELECT s.{string.Join(',', columnNames)} FROM [{configuration.GetSchemaNameOrDefault(mapping)}].[{mapping.TableName}] s INNER JOIN @criteriaTable t on t.[ParameterValue] = s.[{mapping.IdColumn.ColumnName}] order by s.[{mapping.IdColumn.ColumnName}]";
             return new PreparedCommand(statement, param, RetriableOperation.Select, mapping, commandBehavior: CommandBehavior.SingleResult | CommandBehavior.SequentialAccess);
         }
 
-        public string[] GetColumnNames(string? schemaName, string tableName)
+        public string[] GetColumnNames(string schemaName, string tableName)
```

Comment:
Note: I wouldn't consider this a breaking change as it's not breaking at runtime -> nullability is only enforced by the compiler.